crates/flycheck/src/lib.rs

```diff
@@ -329,8 +329,8 @@ impl CargoActor {
             Ok(output) if output.status.success() => Ok(()),
             Ok(output)  => {
                 Err(io::Error::new(io::ErrorKind::Other, format!(
-                    "Cargo watcher failed, the command produced no valid metadata (exit code: {:?})",
-                    output.status
+                    "Cargo watcher failed, the command produced no valid metadata (exit code: {:?})\nCargo's stderr output:\n{}",
+                    output.status, from_utf8(&output.stderr).unwrap_or("(Error: Could not fetch Cargo's stderr output)")
```

Comment:
Maybe use from_utf8_lossy instead? That is infallible. In any case the error should say that the stderr is not valid UTF-8 if you keep using from_utf8.

Reply:
You don't want to use `output.stderr` here, we use `streaming_output` to fill in the error output into the `error` local here which is what contains the error (See the `Err(e)` branch below this one).

Reply:
Thank you for your time @bjorn3 and @Veykril. I have updated my PR, now using `error` String variable to build the final output message.